docs/how-to-guides/develop-a-tool/add-a-tool-icon.md

```diff
@@ -14,40 +14,76 @@ Adding a custom tool icon is optional. If you do not provide one, the system use
   - Avoid complex images with lots of detail or contrast, as they may not resize well. 
 
   See [this example](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/icons/custom-tool-icon.png) as a reference.
-- Install dependencies to generate icon data URI:
 
-  ```
-  pip install pillow
-  ```
+## Add tool icon with _icon_ parameter
 
-## Add tool icon with _icon_ parameter 
-Run the command below in your tool project directory to automatically generate your tool YAML, use _-i_ or _--icon_ parameter to add a custom tool icon:
+### Initialize a package tool with icon
+You can use [pf tool init](../../reference/pf-command-reference.md#pf-tool-init) to initialize a package tool with icon:
+```bash
+pf tool init --package <package-name> --tool <tool-name> --set icon=<icon-path>
 ```
-python <promptflow github repo>\scripts\tool\generate_package_tool_meta.py -m <tool_module> -o <tool_yaml_path> -i <tool-icon-path>
-```
-Here we use [an existing tool project](https://github.com/microsoft/promptflow/tree/main/examples/tools/tool-package-quickstart) as an example.
-```
-cd D:\proj\github\promptflow\examples\tools\tool-package-quickstart
 
-python D:\proj\github\promptflow\scripts\tool\generate_package_tool_meta.py -m my_tool_package.tools.my_tool_1 -o my_tool_package\yamls\my_tool_1.yaml -i my_tool_package\icons\custom-tool-icon.png
+PF CLI will copy the icon file to the folder `<package-name>/icon/<icon-file-name>` and generate a tool script in the package. The tool icon will be configured in the tool script. Here we use [an existing tool project](https://github.com/microsoft/promptflow/tree/main/examples/tools/tool-package-quickstart) as an example, the code is as follows:
```

Comment:
Here we use [an existing tool project]  ->  Here we use [an existing tool](link to my_tool_1.py file)